packages/smooth_app/lib/pages/user_preferences_page.dart

```diff
@@ -227,11 +218,389 @@ class UserPreferencesPage extends StatelessWidget {
+          ),
+        ),
+      );
+
+  String _getPreferenceFlagKey(final int groupIndex) {
+    switch (groupIndex) {
+      case 0:
+        return 'profile';
+      case 1:
+        return 'food';
+      case 2:
+        return 'vrac';
+    }
+    throw Exception('unknown group index: $groupIndex');
+  }
+
+  bool _isCollapsed(
+    final int groupIndex,
+    final UserPreferences userPreferences,
+  ) =>
+      userPreferences.getFlag(_getPreferenceFlagKey(groupIndex)) ??
+      // only group "food" is supposed to be expanded by default at init time
+      groupIndex != 1;
+
+  Future<void> _switchCollapsed(
+    final int groupIndex,
+    final UserPreferences userPreferences,
+  ) async =>
+      userPreferences.setFlag(
+        _getPreferenceFlagKey(groupIndex),
+        !_isCollapsed(
+          groupIndex,
+          userPreferences,
+        ),
+      );
+
+  Widget _getGroupTitle(
+    final int groupIndex,
+    final UserPreferences userPreferences,
+    final ThemeData themeData,
+  ) {
+    final String label;
+    final String subLabel;
+    final bool extra = groupIndex == 0;
+    switch (groupIndex) {
+      case 0:
+        label = 'Your Profile';
+        subLabel = 'Set app settings and find out advices and blah blah';
+        break;
+      case 1:
+        label = 'Food Preferences';
+        subLabel = 'Choose what information about food matters most to you';
+        break;
+      case 2:
+        label = 'App Settings';
+        subLabel = 'Dark mode, country, color, ...';
+        break;
+      default:
+        throw Exception('unknown group index: $groupIndex');
+    }
+    Widget title = Text(label, style: themeData.textTheme.headline2);
+    if (extra) {
+      title = Row(
+        mainAxisAlignment: MainAxisAlignment.start,
+        crossAxisAlignment: CrossAxisAlignment.start,
+        children: [
+          title,
+          Container(
+            width: 15,
+            height: 15,
+            child: Center(
+              child: Text(
+                '1',
+                style: TextStyle(
+                  color: Colors.white,
+                  fontSize: 8,
+                  fontWeight: FontWeight.bold,
+                ),
+              ),
+            ),
+            decoration: BoxDecoration(
+              color: Colors.red,
+              shape: BoxShape.circle,
+            ),
+          ),
+        ],
+      );
+    }
+    return ListTile(
+      title: title,
+      subtitle: Text(subLabel),
+      trailing: Icon(_isCollapsed(groupIndex, userPreferences)
+          ? Icons.expand_more
+          : Icons.expand_less),
+      onTap: () {
+        _switchCollapsed(groupIndex, userPreferences);
+        setState(() {});
+      },
+    );
+  }
+
+  int _getCollapsedSize(
+      final int groupIndex, final List<AttributeGroup> groups) {
+    switch (groupIndex) {
+      case 0:
+        return 5;
+      case 1:
+        return groups.length;
+      case 2:
+        return 3;
```

Comment:
A user items length and settings items length int would certainly be nice for later additions